muduo/net/Channel.h

```diff
@@ -36,7 +36,9 @@ class Channel : noncopyable
   typedef std::function<void()> EventCallback;
   typedef std::function<void(Timestamp)> ReadEventCallback;
 
-  Channel(EventLoop* loop, int fd);
+  typedef std::function<void(int, Timestamp)> EventsCallback;
+
+  Channel(EventLoop* loop, int fd, bool classify = true);
```

Comment:
I suggest not to introduce the new `classify` parameter. Instead, if EventsCallback is provided, it intercepts all events.

Reply:
Do you mean change
  channel_->setEventsCallback(std::bind(&MariaDBClient::stateMachineHandler, this, kRealConnectCont, _1, _2));
to
  channel_->setReadCallback(std::bind(&MariaDBClient::stateMachineHandler, this, kRealConnectCont, _1, _2));
  channel_->setWriteCallback(std::bind(&MariaDBClient::stateMachineHandler, this, kRealConnectCont, _1, _2));
?